File: sdk-dotnet/Examples/ExternalEventExample/WaitForExternalEventWorker.cs
using LittleHorse.Sdk.Worker;

namespace ExternalEventExample;

public class WaitForExternalEventWorker
{
    private int _counter = 0;

    [LHTaskMethod("greet")]
    public async Task<string> Greet(string name)
    {
        Console.WriteLine("Hello there, " + name);

        return "greeted-" + name.ToLower() + "-" + _counter++;
    }
    [LHTaskMethod("summary")]
    public async Task<string> ShowSummary(string name, int age)
    {
        name = name.ToUpper();
        var summary = $"Summary: {name} is {age} years old";
        if (age < 18)
        {
            summary += " and is a minor.";
        }
        else
        {
            summary += " and an adult.";
        }

        return summary;
    }
}

File: examples/java/external-event/src/main/java/io/littlehorse/examples/WaitForExternalEventWorker.java
package io.littlehorse.examples;

import io.littlehorse.sdk.worker.LHTaskMethod;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class WaitForExternalEventWorker {

    private static final Logger log = LoggerFactory.getLogger(WaitForExternalEventWorker.class);

    @LHTaskMethod("ask-for-name")
    public String askForName() {
        log.debug("Executing ask-for-name");
        return "Hi what's your name?";
    }

    @LHTaskMethod("greet")
    public String greet(String name) {
        log.debug("Executing greet");
        return "Hello there, " + name;
    }
}
